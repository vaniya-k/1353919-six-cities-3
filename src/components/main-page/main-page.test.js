import React from 'react';
import renderer from 'react-test-renderer';
import MainPage from './main-page.jsx';
import {createStore} from "redux";
import {Provider} from "react-redux";
import {Router} from 'react-router-dom';
import history from '../../history.js';

const mockReducerWithPlaces = () => {
  const state = {
    offers: {
      activeCityName: `Metropolis`,
      activeCityId: 0,
      activeCardLatLon: {lat: 123, lon: 123},
      activeSortType: 0,
      places: [
        {
          id: 1,
          title: `Solitude Fortress`,
          price: 123,
          type: `house`,
          rating: 100,
          isPremium: true,
          isFavorite: true,
          previewUrl: `https://`,
          gps: {lat: 123, lon: 123}
        },
        {
          id: 2,
          title: `Daily Planet Office`,
          price: 321,
          type: `apartment`,
          rating: 20,
          isPremium: false,
          isFavorite: true,
          previewUrl: `https://`,
          gps: {lat: 123, lon: 123}
        }
      ],
      allOffers: [{
        city: `Metropolis`,
        cityLatLon: {lat: 123, lon: 123},
        places: [
          {
            id: 1,
            title: `Solitude Fortress`,
            price: 123,
            type: `house`,
            rating: 100,
            isPremium: true,
            isFavorite: true,
            previewUrl: `https://`,
            gps: {lat: 123, lon: 123}
          },
          {
            id: 2,
            title: `Daily Planet Office`,
            price: 321,
            type: `apartment`,
            rating: 20,
            isPremium: false,
            isFavorite: true,
            previewUrl: `https://`,
            gps: {lat: 123, lon: 123}
          }
        ]
      },
      {
        city: `Gotham`,
        cityLatLon: {lat: 123, lon: 123},
        places: [
          {
            id: 3,
            title: `Batman's Den`,
            price: 456,
            type: `house`,
            rating: 60,
            isPremium: false,
            isFavorite: true,
            previewUrl: `https://`,
            gps: {lat: 123, lon: 123}
          },
          {
            id: 4,
            title: `Wayne Estate`,
            price: 789,
            type: `apartment`,
            rating: 100,
            isPremium: true,
            isFavorite: true,
            previewUrl: `https://`,
            gps: {lat: 123, lon: 123}
          }
        ]
      },
      {
        city: `Utopia`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      },
      {
        city: `Dystopia`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      },
      {
        city: `Inferno`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      },
      {
        city: `Limbo`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      }]
    },
    user: {
      authorizationStatus: `AUTH`
    }
  };
  return state;
};

const mockStoreWithPlaces = createStore(mockReducerWithPlaces);

const mockReducerWithoutPlaces = () => {
  const state = {
    offers: {
      activeCityName: `Metropolis`,
      activeCityId: 0,
      activeCardLatLon: {lat: null, lon: null},
      activeSortType: 0,
      places: [],
      allOffers: [{
        city: `Metropolis`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      },
      {
        city: `Gotham`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      },
      {
        city: `Utopia`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      },
      {
        city: `Dystopia`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      },
      {
        city: `Inferno`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      },
      {
        city: `Limbo`,
        cityLatLon: {lat: 123, lon: 123},
        places: []
      }]
    },
    user: {
      authorizationStatus: `AUTH`
    }
  };
  return state;
};

const mockStoreWithoutPlaces = createStore(mockReducerWithoutPlaces);

it(`<MainPage/> with Metropolis as the selected city and a couple of places there`, () => {
  const tree = renderer
    .create(<Provider store={mockStoreWithPlaces}><Router history={history}><MainPage/></Router></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`<MainPage/> with Metropolis as the selected city and no places there`, () => {
  const tree = renderer
    .create(<Provider store={mockStoreWithoutPlaces}><Router history={history}><MainPage/></Router></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
