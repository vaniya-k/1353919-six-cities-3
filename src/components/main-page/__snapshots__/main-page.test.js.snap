// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage/> with Metropolis as the selected city and a couple of places there 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Batman
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index null"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Metropolis
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Gotham
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Utopia
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Dystopia
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Inferno
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Limbo
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            2
             places to stay in 
            Metropolis
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by 
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  href="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom null"
            >
              <li
                className="places__option places__option--active"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="cities__places-list places__list tabs__content"
          >
            <article
              className="cities__place-card place-card"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="110"
                    src="https://"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      123
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button place-card__bookmark-button--active button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      In bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/place/1"
                    onClick={[Function]}
                  >
                    Solitude Fortress
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  house
                </p>
              </div>
            </article>
            <article
              className="cities__place-card place-card"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="110"
                    src="https://"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      321
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button place-card__bookmark-button--active button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      In bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "20%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/place/2"
                    onClick={[Function]}
                  >
                    Daily Planet Office
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  apartment
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <div
              id="mapid"
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`<MainPage/> with Metropolis as the selected city and no places there 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Batman
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Metropolis
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Gotham
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Utopia
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Dystopia
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Inferno
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item null"
              href="#"
              onClick={[Function]}
            >
              <span>
                Limbo
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container cities__places-container--empty container"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              No places to stay available
            </b>
            <p
              className="cities__status-description"
            >
              We could not find any property availbale at the moment in 
              Metropolis
            </p>
          </div>
        </section>
        <div
          className="cities__right-section"
        />
      </div>
    </div>
  </main>
</div>
`;
